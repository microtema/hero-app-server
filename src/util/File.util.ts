import readdirSync from 'recursive-readdir-synchronous';

const cleanupModelPath = (it) => it.substring(it.indexOf('/src') + 4).replace('.ts', '');

/**
 *
 * @param filter may be null of undefined
 */
const loadModules = (filter?: any): any[] => {

    return readPaths(filter)
        .map((it) => require(it))
        .map((it) => it.default);
};

/**
 *
 * @param filter may be null or not defined
 * @param prefix may be null or not defined
 */
const readPaths = (filter?: any, prefix?: string): string[] => {

    return readdirSync('./src')
        .filter((it) => it.match(filter || ''))
        .map(cleanupModelPath)
        .map((it) => (prefix || '..') + it);
};

export {loadModules, readPaths};
