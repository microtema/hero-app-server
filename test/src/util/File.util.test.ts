import * as sut from './../../../src/util/File.util';

describe('File Util', () => {

    it('It should be defined', () => {

        expect(sut).toBeDefined();
    });

    it('readPaths', () => {

        expect(sut.readPaths('.util.ts')).toEqual(['../util/File.util']);
    });

    it('readPaths without filter', () => {

        expect(sut.readPaths()).toBeDefined();
    });

    it('readPaths with prefix', () => {

        expect(sut.readPaths('.util.ts', '$')).toEqual(['$/util/File.util']);
    });

    it('loadModules', () => {

        expect(sut.loadModules('Author.schema.ts')).toMatchSnapshot();
    });
});
