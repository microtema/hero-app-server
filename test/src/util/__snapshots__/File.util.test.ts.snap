// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File Util loadModules 1`] = `
Array [
  Object {
    "definitions": Array [
      Object {
        "directives": Array [],
        "fields": Array [
          Object {
            "arguments": Array [
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "id",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "ID",
                  },
                },
              },
            ],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "author",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Author",
              },
            },
          },
          Object {
            "arguments": Array [
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "name",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "String",
                  },
                },
              },
            ],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "authors",
            },
            "type": Object {
              "kind": "ListType",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "Author",
                },
              },
            },
          },
        ],
        "interfaces": Array [],
        "kind": "ObjectTypeExtension",
        "name": Object {
          "kind": "Name",
          "value": "Query",
        },
      },
      Object {
        "directives": Array [],
        "fields": Array [
          Object {
            "arguments": Array [
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "name",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "String",
                  },
                },
              },
            ],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "createAuthor",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Author",
              },
            },
          },
          Object {
            "arguments": Array [
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "id",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "ID",
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "name",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "String",
                  },
                },
              },
            ],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "updateAuthor",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Author",
              },
            },
          },
          Object {
            "arguments": Array [
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "id",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "ID",
                  },
                },
              },
            ],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "deleteAuthor",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Boolean",
              },
            },
          },
        ],
        "interfaces": Array [],
        "kind": "ObjectTypeExtension",
        "name": Object {
          "kind": "Name",
          "value": "Mutation",
        },
      },
      Object {
        "description": undefined,
        "directives": Array [],
        "fields": Array [
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "id",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "ID",
              },
            },
          },
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "name",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "String",
              },
            },
          },
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "books",
            },
            "type": Object {
              "kind": "ListType",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "Book",
                },
              },
            },
          },
        ],
        "interfaces": Array [],
        "kind": "ObjectTypeDefinition",
        "name": Object {
          "kind": "Name",
          "value": "Author",
        },
      },
    ],
    "kind": "Document",
    "loc": Object {
      "end": 350,
      "start": 0,
    },
  },
]
`;
